src/main/java/core/basesyntax/strategy/OperationHandlerFactory.java

```diff
@@ -0,0 +1,24 @@
+package core.basesyntax.strategy;
+
+import core.basesyntax.model.Transaction;
+import core.basesyntax.service.OperationHandler;
+import core.basesyntax.service.StorageService;
+import core.basesyntax.service.impl.StorageServiceImpl;
+import java.util.HashMap;
+import java.util.Map;
+
+public class OperationHandlerFactory {
+    private final Map<Transaction.Operation, OperationHandler> operationHandlers = new HashMap<>();
+    private final StorageService storageService = new StorageServiceImpl();
+
+    public OperationHandlerFactory() {
+        operationHandlers.put(Transaction.Operation.S, new IncrementHandler(storageService));
+        operationHandlers.put(Transaction.Operation.R, new IncrementHandler(storageService));
+        operationHandlers.put(Transaction.Operation.P, new DecrementHandler(storageService));
+        operationHandlers.put(Transaction.Operation.B, new IncrementHandler(storageService));
```

Comment:
According to the checklist, avoid using switch-cases and if-else constructions. It is recommended to use the Strategy pattern instead. Here you have correctly implemented the Strategy pattern by mapping operations to their respective handlers, which is good. However, make sure that the strategies for 'S' and 'B' operations are correct, as usually 'B' stands for balance and might not require an increment operation like 'S' and 'R'.